boards/BMS/Src/App/states/App_ChargeState.c

```diff
@@ -25,18 +34,42 @@ static void ChargeStateRunOnTick100Hz(struct StateMachine *const state_machine)
     struct BmsWorld *world = App_SharedStateMachine_GetWorld(state_machine);
     struct BmsCanTxInterface *can_tx  = App_BmsWorld_GetCanTx(world);
     struct Charger *          charger = App_BmsWorld_GetCharger(world);
+    struct Clock *            clock   = App_BmsWorld_GetClock(world);
 
     if (!App_Charger_IsConnected(charger))
     {
         App_CanTx_SetPeriodicSignal_CHARGER_DISCONNECTED_IN_CHARGE_STATE(
             can_tx, true);
         App_SharedStateMachine_SetNextState(state_machine, App_GetFaultState());
     }
+
+    if (App_Charger_HasFaulted(charger))
+    {
+        App_CanTx_SetPeriodicSignal_CHARGER_FAULT_DETECTED(can_tx, true);
+        App_SharedStateMachine_SetNextState(state_machine, App_GetFaultState());
+    }
+
```

Comment:
Most of the if statements here direct us to the fault state. In some cases certain sections of code do not need to be executed either. See if we can make this nicer :)

Reply:
to be more specific, we dont have to check for elapsed time when the charger has faulted or if the charger is not connected